File: app/src/main/java/com/bottle/track/user/BaseRequestBean.java
package com.bottle.track.user;

import com.bottle.track.api.BaseInfoBean;

/**
 * Created by Administrator on 2017/12/12 0012.
 */

public class BaseRequestBean<T> {
    private BaseInfoBean baseInfoBean;
    private T requestBody;


     BaseRequestBean(T requestBody){
        this.requestBody = requestBody;
        this.baseInfoBean = new BaseInfoBean();
    }
    public BaseInfoBean getBaseInfoBean() {
        return baseInfoBean;
    }

    public void setBaseInfoBean(BaseInfoBean baseInfoBean) {
        this.baseInfoBean = baseInfoBean;
    }

    public T getRequestBody() {
        return requestBody;
    }

    public void setRequestBody(T requestBody) {
        this.requestBody = requestBody;
    }
}

File: app/src/main/java/com/bottle/track/api/BaseRequestBean.kt
package com.bottle.track.api

/**
 * @ClassName BaseRequestBean
 * @Author half_bottle
 * @Date 2017/12/1
 * @Description POST请求的body
 */
class BaseRequestBean<T>{

    val baseinfo: BaseInfoBean
    val reqbody: T

    constructor(requestBody: T){
        this.baseinfo = BaseInfoBean()
        this.reqbody = requestBody
    }

}